runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformTranslation.java

```diff
@@ -196,10 +196,13 @@ private PTransformTranslation() {}
         transformBuilder.setSpec(spec);
       }
     } else if (KNOWN_PAYLOAD_TRANSLATORS.containsKey(transform.getClass())) {
-      transformBuilder.setSpec(
+      FunctionSpec spec =
           KNOWN_PAYLOAD_TRANSLATORS
               .get(transform.getClass())
-              .translate(appliedPTransform, components));
+              .translate(appliedPTransform, components);
+      if (spec != null) {
```

Comment:
It would be preferable to return an empty spec above than add this check here. (Unless there's code that actually branches on missing differently than empty).

Reply:
There is, setting the spec to null causes an exception, whereas leaving it missing is fine.

Reply:
This follows the pattern a few lines higher where `null` function specs are used as a signal to represent that they are absent.